src/Edge.ts

```diff
@@ -0,0 +1,39 @@
+import Vector, {execute as vectorExecute} from "./Vector";
+import {Graph, SchedulerEvent, newId, EdgeError, Connector} from "./Shared";
+import Scheduler from "./Scheduler";
+/** The edge of a vector, what connectors connect to. */
+export default interface Edge {
+    /** Name of the edge */
+    field: string;
+    /** Connectors that connect the edges together */
+    connectors: Connector[];
+}
+/** Executes a given edge.  Edges are always inputs (LTR) */
+export async function execute(scheduler: Scheduler, graph: Graph, vector: Vector, field: string, value: any): Promise<any> {
+    scheduler.dispatchEvent("beginedge", {
+        time: Date.now(),
+        id: newId(),
+    } as SchedulerEvent);
+    const start = Date.now();
```

Comment:
@TonyGermaneri nit: Date.now() called twice when you could use the cached var `start` in your dispatchEvent json